### PR TITLE
Move Opinions/Compatibility -> Design Decisions

### PR DESCRIPTION
**Pull request by ayuishii**
Resolve issues for Opinions & Compatibility and move this section Design Decisions in the explainer to keep the section concise. Alternatively I can move this section to a new `Appendix` section at the bottom of the spec, similar to Fetch's [Background reading](https://fetch.spec.whatwg.org/#background-reading).
LMK what you think.

***
<a href="https://pr-preview.s3.amazonaws.com/ayuishii/cookie-store/pull/174.html" title="Last updated on Sep 15, 2020, 10:50 PM UTC (f5ead24)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/cookie-store/174/5a58687...ayuishii:f5ead24.html" title="Last updated on Sep 15, 2020, 10:50 PM UTC (f5ead24)">Diff</a>

### CODE REVIEW

File: explainer.md
Comment (pwnall) [inline explainer.md:541]: Regarding "Cookies with an empty name cannot be set..." -- I vaguely remember that we changed Chrome's implementation of this spec to completely disallow writing cookies with empty names.

I think we should lift this behavior into the spec and outright disallow setting cookies with empty names, if we haven't already done so by virtute of WPT.

File: explainer.md
Comment (pwnall) [inline explainer.md:546]: The text in this section feels redundant with use cases and goals.

File: explainer.md
Comment (pwnall) [inline explainer.md:550]: "Scripts should not have to write and then read "test cookies" to determine whether script-initiated cookie write access is possible" -- I read this as providing some `canWriteCookie` API, which we haven't.

I'm guessing this entire section aims to say that we choose to fail explicitly on writing instead of silently like `document.cookie` does. I don't think this qualifies as an opinion, as this behavior is consistent with every modern storage API.

File: explainer.md
Comment (pwnall) [inline explainer.md:552]: nit: Compatability -> Compatibility

File: explainer.md
Comment (pwnall) [inline explainer.md:557]: This paragraph feels like a duplicate of the "Interpreting Strings" subsection above. Merge into that section?

File: explainer.md
Comment (pwnall) [inline explainer.md:560]: In practice this will change the behavior of -> this mandates changes to the behavior of...

**Review by pwnall — CHANGES_REQUESTED** [review]
Really solid start! Thank you for doing this!

File: explainer.md
Comment (ayuishii) [reply explainer.md:530]: I see, this was an intentional change from this [issue](https://github.com/WICG/cookie-store/issues/165), but maybe I should be versioning links that point to a specific section.

File: explainer.md
Comment (ayuishii) [reply explainer.md:541]: I think currently setting empty names is still allowed ([example](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/external/wpt/cookie-store/cookieStore_delete_arguments.tentative.https.any.js;l=163?q=empty%20name&ss=chromium%2Fchromium%2Fsrc:third_party%2Fblink%2Fweb_tests%2Fexternal%2Fwpt%2Fcookie-store%2F&start=1)), and only disallowed if the value has an `=` as it mentions above ([example](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/external/wpt/cookie-store/cookieStore_set_arguments.tentative.https.any.js;l=34)). Do you think this should be returning a `TypeError` in the case of an empty name?

File: explainer.md
Comment (ayuishii) [reply explainer.md:533]: SG, will make an AI for myself to add this change.

File: explainer.md
Comment (ayuishii) [reply explainer.md]: Updated links

File: explainer.md
Comment (pwnall) [reply explainer.md:570]: Thank you for explaining! I defer to the cookie experts who reviewed our implementation.

**Review by pwnall — APPROVED** [review]
LGTM, thank you for the revisions!

**Comment by ayuishii** [comment]
Thanks for the review!

**Pull request merged**
